@here/harp-mapview/lib/TileProxy.ts

```diff
@@ -0,0 +1,268 @@
+     */
+    constructor(readonly tile: Tile, readonly offset: number) {
+        assert(!tile.isProxy);
+    }
+
+    addUserTextElement(textElement: TextElement): void {
+        this.tile.addUserTextElement(textElement);
+    }
+
+    removeUserTextElement(textElement: TextElement): boolean {
+        return this.removeUserTextElement(textElement);
+    }
+
+    addTextElement(textElement: TextElement): void {
+        this.addTextElement(textElement);
+    }
+
+    removeTextElement(textElement: TextElement): boolean {
+        return this.removeTextElement(textElement);
+    }
+
+    prepareForRender(): void {
+        this.tile.prepareForRender();
+    }
+
+    willRender(_zoomLevel: number): boolean {
+        return this.tile.willRender(_zoomLevel);
+    }
+
+    didRender(): void {
+        this.tile.didRender();
+    }
+
+    shouldDisposeObjectGeometry(object: TileObject): boolean {
+        return this.shouldDisposeObjectGeometry(object);
+    }
+
+    shouldDisposeObjectMaterial(object: TileObject): boolean {
+        return this.shouldDisposeObjectMaterial(object);
+    }
+
+    forceHasGeometry(value: boolean): void {
+        this.tile.forceHasGeometry(value);
+    }
+
+    /**
+     * Clears internal cleared items and asks the tile to reload.
+     */
+    reload(): void {
+        this.dispose();
+        this.tile.reload();
+    }
+
+    /**
+     * Clears cloned items. We don't proxy the [[Tile]]s dispose method because this will be called
+     * for that particular object (remember this object doesn't own the [[Tile]]), we just clear the
+     * items we have cloned ourself.
+     */
+    dispose(): void {
+        this.objects = [];
+        this.geometryVersion = 0;
+    }
+
+    prepareTextPaths(textPathGeometries: TextPathGeometry[]): TextPathGeometry[] {
+        return this.tile.prepareTextPaths(textPathGeometries);
+    }
+
+    createTextElements(decodedTile: DecodedTile): void {
+        this.tile.createTextElements(decodedTile);
+    }
+
+    /**
+     * Clones the underlying [[Tile]] TileObjects. The shifting of the positions is handled in the
+     * [[renderTileObjects]] function in [[MapView]].
+     */
+    clone() {
+        assert(this.geometryVersion !== this.tile.geometryVersion);
+        assert(this.tile.hasGeometry);
+        for (const tileObject of this.tile.objects) {
+            // See: https://github.com/mrdoob/three.js/pull/15062
+            if (!(tileObject instanceof THREE.Line)) {
+                const clone = tileObject.clone();
+                // These aren't cloned across unfortunately and must be manually transferred.
+                clone.onBeforeRender = tileObject.onBeforeRender;
+                clone.onAfterRender = tileObject.onAfterRender;
+                this.objects.push(clone);
+            }
+        }
+        // Todo: Clone text elements also.
+        this.geometryVersion = this.tile.geometryVersion;
+    }
+
+    /**
+     * Checks if the [[TileProxy]] can be cloned.
```

Comment:
Actually "should be cloned" ....
...

On the other hand, why is this even public ?

Shouldn't `TileProxy.prepareForRender` check if it's geometry copy is up to date ?

Reply:
Clone is done in VisibleTileSet, hence being public.

However this could be a good alternative, I will take a look.